tests/Model/StubsContainer.php

```diff
@@ -69,13 +70,24 @@ public function addFunction(PHPFunction $function): void
         }
     }
 
-    #[Pure]
-    public function getClass(string $name): ?PHPClass
+    /**
+     * @param string $name
+     * @param string|null $sourceFilePath
+     * @return PHPClass|null
+     * @throws RuntimeException
+     */
+    public function getClass(string $name, ?string $sourceFilePath = null): ?PHPClass
     {
-        if (array_key_exists($name, $this->classes) && isset($this->classes[$name])) {
-            return $this->classes[$name];
+        $classes = array_filter($this->classes, fn(PHPClass $class): bool => $class->name === $name);
```

Comment:
Let's preserve backward compatibility and return an element if it's the only one. The `sourceFilePath` based resolve will kick in only if we find duplicates.